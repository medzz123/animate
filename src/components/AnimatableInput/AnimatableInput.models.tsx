export interface AnimatableInputProps
  extends React.InputHTMLAttributes<HTMLInputElement> {
  /**
   * Label to display next to the text input.
   */
  label: string;
}
