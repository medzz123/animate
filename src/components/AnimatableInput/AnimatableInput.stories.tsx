// import { Meta, Story } from '@storybook/react/types-6-0';
// import React from 'react';

// import AnimatableInput from './AnimatableInput';

// export default {
//   title: 'Components/AnimatableInput',
//   component: AnimatableInput,
// } as Meta;

// const Template: Story = (storyArguments) => (
//   <AnimatableInput label="Story Input" {...storyArguments} />
// );

// export const AnimatableInputStory = Template.bind({});

// AnimatableInputStory.args = {
//   placeholder: '10rem 20rem',
// };
