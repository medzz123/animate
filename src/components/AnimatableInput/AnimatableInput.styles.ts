import styled from 'styled-components';

import { tokens } from '../../theme/tokens';

export const AnimatableInputContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;

  max-width: ${tokens.sizes['2xs']};
  margin-bottom: ${tokens.sizes[12]};
`;

export const Label = styled.label`
  font-size: ${tokens.sizes[12]};
  color: ${(p) => p.theme.somethingNew.flip};
  flex: 1;
  margin-right: ${tokens.sizes[8]};
  min-width: ${tokens.sizes[100]};
`;

export const StyledInput = styled.input`
  font-size: ${tokens.sizes[12]};
  padding: ${tokens.sizes[5]};
  color: ${(p) => p.theme.somethingNew.flip};
  border: none;
  box-shadow: inset 0 0 0 ${tokens.sizes[1]} ${(p) => p.theme.somethingNew.flip};
  background-color: transparent;
  width: ${tokens.sizes.full};
  flex: 2;

  ::placeholder {
    color: ${(p) => p.theme.somethingNew.flip};
    opacity: 0.5;
  }

  border-radius: ${tokens.sizes[8]};

  &:focus {
    outline: none;
  }
`;
