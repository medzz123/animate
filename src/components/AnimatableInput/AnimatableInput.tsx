import React, { FunctionComponent } from 'react';

import { AnimatableInputProps } from './AnimatableInput.models';
import {
  AnimatableInputContainer,
  Label,
  StyledInput,
} from './AnimatableInput.styles';

const AnimatableInput: FunctionComponent<AnimatableInputProps> = (props) => {
  const { label, ...rest } = props;

  return (
    <AnimatableInputContainer>
      <Label htmlFor={label} data-testid="animatable-input-label">
        {label}
      </Label>
      <StyledInput
        id={label}
        autoComplete="off"
        data-testid="animatable-input"
        {...rest}
      />
    </AnimatableInputContainer>
  );
};

export default AnimatableInput;
