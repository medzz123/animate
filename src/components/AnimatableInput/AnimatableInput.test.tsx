import { fireEvent, render } from '@test/testUtils';
import React from 'react';

import AnimatableInput from './AnimatableInput';

test('AnimatableInput Test', () => {
  const { getByTestId, container } = render(<AnimatableInput label="Test" />);

  const animatableInputNode = getByTestId(
    'animatable-input'
  ) as HTMLInputElement;

  const animatableInputLabelNode = getByTestId('animatable-input-label');

  fireEvent.change(animatableInputNode, { target: { value: 'Hello' } });

  expect(animatableInputNode.value).toBe('Hello');
  expect(animatableInputLabelNode.textContent).toBe('Test');

  expect(container).toMatchSnapshot();
});
