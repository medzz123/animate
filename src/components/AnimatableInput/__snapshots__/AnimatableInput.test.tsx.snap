// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatableInput Test 1`] = `
<div>
  <div
    class="sc-bdfBwQ cMLcsh"
  >
    <label
      class="sc-gsTCUz enBbRt"
      data-testid="animatable-input-label"
      for="Test"
    >
      Test
    </label>
    <input
      autocomplete="off"
      class="sc-dlfnbm fQvILG"
      data-testid="animatable-input"
      id="Test"
    />
  </div>
</div>
`;
